src/acquisition/covidcast/csv_importer.py

```diff
@@ -271,13 +278,62 @@ def extract_and_check_row(row, geo_type):
 
     # optional not-too-small float
     try:
+      # Returns either a float, None, or causes an exception
       sample_size = CsvImporter.maybe_apply(float, row.sample_size)
     except ValueError:
       # expected a number, but got a string
       return (None, 'sample_size')
 
+    if hasattr(row, "missing_val"):
+      missing_value = row.missing_val
+      # not an int code
+      if not isinstance(missing_value, int):
+        # capture a few possible ways to write a nan
+        if str(missing_value).lower() in {'na', 'nan', 'none', ''}:
+          missing_value = 0
+        # if not an int or a nan, then we have problems
+        else:
+          return (None, 'missing_val')
+      # missing code contradicts the value being present
+      elif value is not None and missing_value > 0:
+        return (None, 'missing_val')
+    else:
+      missing_value = 0
+
+    if hasattr(row, "missing_se"):
+      missing_stderr = row.missing_se
+      # not an int code
+      if not isinstance(missing_stderr, int):
+        # capture a few possible ways to write a nan
+        if str(missing_stderr).lower() in {'na', 'nan', 'none', ''}:
+          missing_stderr = 0
+        # if not an int or a nan, then we have problems
+        else:
+          return (None, 'missing_se')
+      # missing code contradicts the value being present
+      if stderr is not None and missing_stderr > 0:
+        return (None, 'missing_se')
+    else:
+      missing_stderr = 0
+
+    if hasattr(row, "missing_sample_size"):
```

Comment:
It looks like this code is mostly replicated 3 times -- is there a way to extract the common stuff into a function?